### PR TITLE
Feature/icky 313

### PR DESCRIPTION
**Pull request by kibo-susantasamanta**
@kibo-kevinwatts @kibo-sushant @kibo-chandradeeptalaha

### CODE REVIEW

File: composables/useSearchSuggestions.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { getSearchSuggestions } from "~~/gql/queries"
+import { useState, useNuxtApp } from "#app"
+import * as GraphQLTypes from "@/server/types/GraphQL"
+
+export const useSearchSuggestions = () => {
+  const nuxt = useNuxtApp()
+  const fetcher = nuxt.nuxt2Context.$gqlFetch
+  const result = useState(`use-result-search`, (): GraphQLTypes.SearchSuggestionResult => {
```
Comment (kibo-kevinwatts) [inline composables/useSearchSuggestions.ts]: rename keys so they all contain `search-suggestions`

File: components/KiboAppHeader.vue [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    const { setTermForUrl, getFacetsFromURL, getCatLink } = useUiHelpers()
+    const { result, search, loading } = useSearchSuggestions()
+    const nuxt = useNuxtApp()
+    const app = nuxt.nuxt2Context.app
+    const { user, load: loadUser } = useUser()
+    const searchValue = ref("")
+    const isAuthenticated = computed(() => {
+      return userGetters.isLoggedInUser(user.value)
+    })
+    const term = ref(getFacetsFromURL().phrase)
```
Comment (kibo-kevinwatts) [inline components/KiboAppHeader.vue]: for search suggestion, we don't need to do anything with the URL. 
Query Term should only be from the search bar input as user types

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: composables/useSearchSuggestions.ts (hunk 1 above)
Comment (kibo-susantasamanta) [reply composables/useSearchSuggestions.ts]: Renamed

File: components/KiboAppHeader.vue (hunk 2 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue]: query term from URL removed.

File: components/KiboAppHeader.vue [hunk 3]
```diff
@@ -106,15 +138,61 @@ export default defineComponent({
...
 
+    const closeSearch = () => {
+      if (!isSearchOpen.value) return
+
+      term.value = ""
+      isSearchOpen.value = false
+    }
+
+    const handleSearch = debounce(async (paramValue) => {
+      loading.value = true
```
Comment (kibo-kevinwatts) [inline components/KiboAppHeader.vue:159]: the composable should be updating this loading state

File: components/KiboAppHeader.vue [hunk 4]
```diff
@@ -18,11 +18,38 @@
...
-            placeholder="Search for items"
+            ref="searchBarRef"
+            v-click-outside="closeSearch"
+            :placeholder="$t('Search for items')"
             aria-label="Search"
             class="sf-header__search"
-          />
+            :value="term"
+            @input="handleSearch"
+            @keydown.enter="handleSearch($event)"
```
Comment (kibo-kevinwatts) [inline components/KiboAppHeader.vue:28]: On enter we want to clear the searchbar input, close the Search Suggestions and redirect to the /search page

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/KiboAppHeader.vue (hunk 3 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue:159]: Resolved with icky-305

File: components/KiboAppHeader.vue (hunk 4 above)
Comment (kibo-susantasamanta) [reply components/KiboAppHeader.vue:28]: resolved with icky-305

File: composables/useUiHelpers.ts [hunk 5]
```diff
@@ -1,37 +1,65 @@
...
-    }
+      sort: query.sort,
+    } as getFacetsFromURLResponse
+  }
+
+  const setTermForUrl = (term: string) => {
+    instance.router.push({
+      query: {
+        ...getFiltersDataFromUrl(instance, false),
+        phrase: term || undefined,
```
Comment (kibo-chandradeeptalaha) [inline composables/useUiHelpers.ts:46]: no need to add undefined

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**